service/collector.go

```diff
@@ -100,16 +89,10 @@ func New(set CollectorSettings) (*Collector, error) {
 	}
 
 	return &Collector{
-		set:          set,
-		stateChannel: make(chan State, Closed+1),
+		set: set,
 	}, nil
 }
 
-// GetStateChannel returns state channel of the collector server.
-func (col *Collector) GetStateChannel() chan State {
```

Comment:
Why is this deleted? It is used in the tests.

Reply:
Apologies for not mentioning this on the PR. This channel, given that it is used only in tests and has a size of 3, fills up after a couple of changes in the config and blocks the code flow when we try to add another state in. So i removed it to test and seek clarity on how to move forward.

Reply:
Please create a separate PR if this causes problems :)

Reply:
https://github.com/open-telemetry/opentelemetry-collector/pull/4460 -> done